snappy_wrappers/wrappers/somatic_variants_qc/summarize-vcf.py

```diff
@@ -0,0 +1,269 @@
+                # Need to check multi allelic. Users shouldn't input multi allelic vcf file.
+                if get_variant_type(variant.REF, variant.ALT[0]) == "snp":
+                    n_snps += 1
+                    mt_classes = assign_class_snvs(variant, mt_classes)
+                else:
+                    # More for indels
+                    n_indels += 1
+                    indels_length.append(abs(len(variant.REF) - len(variant.ALT[0])))
+
+        return (
+            total_v_count,
+            v_inside_exom,
+            v_outside_exom,
+            n_snps,
+            n_indels,
+            indels_length,
+            minimal_rp_snvs_exom,
+            minimal_rp_snvs_nexom,
+            limited_rp_snvs_exom,
+            limited_rp_snvs_nexom,
+            strong_rp_snvs_exom,
+            strong_rp_snvs_nexom,
+            mt_classes,
+            vaf,
+        )
+    else:
+        for variant in vcf_file:
+            total_v_count += 1
+        return total_v_count
+
+
+def main():
+    classes = ["A>C", "A>G", "A>T", "C>G", "C>T", "G>T"]
+    path_full_vcf = options.rawvcf
+    path_filtered_vcf = options.filtered_vcf
+    path_bedfile = options.exom_bedfile
+    outpath = options.output
+    full_vcf = VCF(path_full_vcf)
+    filter_vcf = VCF(path_filtered_vcf)
+    # header = filter_vcf.raw_header
+    sample = filter_vcf.samples[1]
+    # read bed file
+    bed_intervals = tabix.open(path_bedfile)
+    full_v_count = process_vcf_file(full_vcf)
+    (
+        passed_v_count,
+        v_inside_exom,
+        v_outside_exom,
+        n_snps,
+        n_indels,
+        indels_length,
+        minimal_rp_snvs_exom,
+        minimal_rp_snvs_nexom,
+        limited_rp_snvs_exom,
+        limited_rp_snvs_nexom,
+        strong_rp_snvs_exom,
+        strong_rp_snvs_nexom,
+        mt_classes,
+        vaf,
+    ) = process_vcf_file(filter_vcf, bed_intervals, filter_file=True, padding=options.padding)
+    # After one for loop through vcf file. The VCF will automatically be closed.
+    # length_exoms = calculate_exoms_length(bed_intervals)
+    # for variant in filter_vcf:
+    #     print(variant.format("AF"))
+    summary = {
+        "sample": sample,
+        "number_full_variants": full_v_count,
+        "number_passed_variants": passed_v_count,
+        "number_snvs_inside_exom": v_inside_exom,
+        "number_snvs_outside_exom": v_outside_exom,
+        "number_snps": n_snps,
+        "number_indels": n_indels,
+        "indels_length": indels_length,
+        "minimal_rp_snvs_exom": minimal_rp_snvs_exom,
+        "minimal_rp_snvs_nexom": minimal_rp_snvs_nexom,
+        "limited_rp_snvs_exom": limited_rp_snvs_exom,
+        "limited_rp_snvs_nexom": limited_rp_snvs_nexom,
+        "strong_rp_snvs_exom": strong_rp_snvs_exom,
+        "strong_rp_snvs_nexom": strong_rp_snvs_nexom,
+        "mutation_classes": mt_classes,
+        "classes": classes,
+        "VAF": vaf,
```

Comment:
Note that if you add all VAFs to the output json file, it can lead to very long lines. A full vcf from WGS data might contain almost one million variants.